management/database/client.go

```diff
@@ -0,0 +1,154 @@
+package database
+
+import (
+	"encoding/xml"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/management"
+)
+
+// Definitions of numerous constants representing API endpoints.
+const (
+	azureCreateDatabaseServerUrl = "services/sqlservers/servers"
+	azureListDatabaseServersUrl  = "/services/sqlservers/servers"
+	azureDeleteDatabaseServerUrl = "/services/sqlservers/servers/%s"
+
+	azureCreateDatabaseUrl = "services/sqlservers/servers/%s/databases"
+	azureGetDatabaseUrl    = "/services/sqlservers/servers/%s/databases/%s"
+	azureListDatabasesUrl  = "/services/sqlservers/servers/%s/databases?contentview=generic"
+	azureDeleteDatabaseUrl = "/services/sqlservers/servers/%s/databases/%s"
+
+	errParamNotSpecified = "Parameter %s is not specified."
+)
+
+// DatabaseClient defines various database CRUD operations.
+// It contains a management.Client for making the actual http calls.
+type DatabaseClient struct {
+	mgmtClient management.Client
+}
+
+// NewClient returns a new DatabaseClient struct with the provided
+// management.Client as the underlying client.
+func NewClient(mgmtClient management.Client) DatabaseClient {
+	return DatabaseClient{mgmtClient}
+}
+
+// CreateDatabaseServer issues a database server creation request to Azure.
+//
+// https://msdn.microsoft.com/en-us/library/azure/dn505699.aspx
+func (c DatabaseClient) CreateDatabaseServer(params DatabaseServerCreateParams) error {
+	if params.Location == "" {
```

Comment:
no need to make these parameter checks unless they're required in constructing the URL, let the API fail with an error.